pkg/v3/resource/certconfig/desired.go

```diff
@@ -101,10 +111,18 @@ func prepareClusterConfig(baseClusterConfig cluster.Config, clusterGuestConfig v
 	if err != nil {
 		return cluster.Config{}, microerror.Mask(err)
 	}
+	clusterConfig.Domain.CalicoEtcdClient, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.CalicoEtcdClientCert)
+	if err != nil {
+		return cluster.Config{}, microerror.Mask(err)
+	}
 	clusterConfig.Domain.Etcd, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.EtcdCert)
 	if err != nil {
 		return cluster.Config{}, microerror.Mask(err)
 	}
+	clusterConfig.Domain.FlanneldEtcdClient, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.FlanneldEtcdClientCert)
+	if err != nil {
+		return cluster.Config{}, microerror.Mask(err)
+	}
```

Comment:
Does it work when is not kvm env?

Reply:
Yes, this works. This is only string interpolation with DNS zone and cert name.